src/Umbraco.Cms.ManagementApi/ManagementApiComposer.cs

```diff
@@ -207,13 +232,22 @@ string ActionSortKeySelector(ApiDescription apiDesc)
                         endpoints.MapControllers();
 
                         // Serve contract
-                        endpoints.MapGet($"{officePath}/management/api/openapi.json",async  context =>
+                        endpoints.MapGet($"{officePath}/management/api/openapi.json", async context =>
                         {
-                            await context.Response.SendFileAsync(new EmbeddedFileProvider(this.GetType().Assembly).GetFileInfo("OpenApi.json"));
+                            await context.Response.SendFileAsync(new EmbeddedFileProvider(GetType().Assembly).GetFileInfo("OpenApi.json"));
                         });
                     });
                 }
             ));
         });
     }
+
+    [GeneratedRegex(".*?\\/v[1-9]+/")]
+    private static partial Regex VersionPrefixRegex();
+
+    [GeneratedRegex("\\{(.*?)\\:?\\}")]
+    private static partial Regex TemplatePlaceholdersRegex();
+
+    [GeneratedRegex("[\\/\\-](\\w{1})")]
+    private static partial Regex ToCamelCaseRegex();
```

Comment:
I'm guessing the partial is because of this, are there some smart magic I'm missing here with the whole partial/attribute thing? Or can we just do this: 

```c#
private static Regex VersionPrefixRegex = new(".*?\\/v[1-9]+/");

private static Regex TemplatePlaceholdersRegex = new("\\{(.*?)\\:?\\}");

private static Regex ToCamelCaseRegex = new ("[\\/\\-](\\w{1})");
```

And omit the whole partial thing?

Reply:
This was mainly because of a suggested thing from Visual Studio. To my understanding, you do it like this to generate the regex at compile-time rather than runtime. It optimizes the regex expression, see more here for [interpreted vs compiled regex](https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices#interpreted-vs-compiled-regular-expressions).

If it hurts the readability of this class, we could probably move the regex'es somewhere else.

Reply:
[Also some more info here](https://devblogs.microsoft.com/dotnet/regular-expression-improvements-in-dotnet-7/#source-generation). It's a feature of .NET 7.

Reply:
Oh would you look at that, the more you know :D But that makes good sense why we're using this to me then, seems like a cool new feature, thanks :D 

But yes as you eluded to I found the readability a bit lacking here, and it just was a bit weird to me that our composer is partial, that doesn't seem right, also since this is bleeding edge, so my IDE complained about it 😉 

But I've made a pr (#13410) that refactors this into its own class and pointed it to this PR, if you're fine with those changes I think we can merge them into this PR and then get this merged 🎉

Reply:
What a great PR, thanks! I approved it, so let's merge when the actions complete!